File: app/src/debug/java/fi/kaupunkifillarit/inject/DebugKaupunkifillaritComponent.java
package fi.kaupunkifillarit.inject;

import dagger.Component;
import fi.kaupunkifillarit.DebugKaupunkifillaritApplication;

@ApplicationScope
@Component(modules = {DebugServicesModule.class, SystemServicesModule.class})
public interface DebugKaupunkifillaritComponent extends KaupunkifillaritComponent {
    /**
     * An initializer that creates the graph from an application.
     */
    final class Initializer {
        public static DebugKaupunkifillaritComponent init(DebugKaupunkifillaritApplication app) {
            return DaggerDebugKaupunkifillaritComponent.builder()
                    .systemServicesModule(new SystemServicesModule(app))
                    .debugServicesModule(new DebugServicesModule(app))
                    .build();
        }

        private Initializer() {
        } // No instances.
    }
}

File: app/src/debug/kotlin/fi/kaupunkifillarit/inject/DebugKaupunkifillaritComponent.kt
package fi.kaupunkifillarit.inject

import dagger.Component
import fi.kaupunkifillarit.DebugKaupunkifillaritApplication
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(DebugServicesModule::class, SystemServicesModule::class))
interface DebugKaupunkifillaritComponent : KaupunkifillaritComponent {
    /**
     * An initializer that creates the graph from an application.
     */
    object Initializer {
        fun init(application: DebugKaupunkifillaritApplication): DebugKaupunkifillaritComponent {
            return DaggerDebugKaupunkifillaritComponent.builder()
                    .systemServicesModule(SystemServicesModule(application))
                    .debugServicesModule(DebugServicesModule(application))
                    .build()
        }
    }
}
